packages/connect/e2e/__txcache__/gen-reftx.js

```diff
@@ -1,4 +1,5 @@
 const BitcoinJs = require('@trezor/utxo-lib');
+const bufferUtils = require('@trezor/utils/lib/bufferUtils');
```

Comment:
I am curious, why are we using imports from `lib/specificFunction`? I am trying to find benefits but these are only use cases I can think about:

1. Someone uses connect in env where it's bundled by webpack or something - every unused function is removed even with dumbest treeshake algorithm because we have `sideEffects: false` in `package.json`
2. Someone uses this in Node env - the whole package is fetched anyway to `node_modules`

Am I missing some use cases?

Reply:
treeshaking is not working if you export everything thru index

expample using connect-iframe package, im changing src/index into:

```
import { bufferUtils } from '@trezor/utils';

console.log(bufferUtils);
```

result:
- **whole** @trezor/utils lib is bundled in build/js/iframe.xxx.js
```
(()=>{var t={"../utils/lib/arrayDistinct.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.arrayDistinct=void 0;e.arrayDistinct=(t,e,r)=>r.indexOf(t)===e},"../utils/lib/arrayPartition.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.arrayPartition=void 0;e.arrayPartition=(t,e)=>t.reduce((([t,r],n)=>e(n)?[[...t,n],r]:[t,[...r,n]]),[[],[]])},"../utils/lib/arrayToDictionary.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.arrayToDictionary=void 0;e.arrayToDictionary=(t,e)=>t.reduce(((t,r)=>Object.assign(Object.assign({},t),{[e(r)]:r})),{})},"../utils/lib/bufferUtils.js":(t,e,r)=>{"use strict";var Buffer=r("../../node_modules/buffer/index.js").lW;Object.defineProperty(e,"__esModule",{value:!0}),e.reverseBuffer=void 0;e.reverseBuffer=t=>{if(t.length<1)return t;const e=Buffer.alloc(t.length);let r=e.length-1;for(let n=0;n<e.length/2;n++)e[n]=t[r],e[r]=t[n],r--;return e}},"../utils/lib/bytesToHumanReadable.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.bytesToHumanReadable=void 0;const r=["B","KB","MB","GB","TB"];e.bytesToHumanReadable=t=>{let e=Math.abs(t),n=0;for(;e>=1024||n>=r.length;)e/=1024,n++;return`${e.toFixed(1)} ${r[n]}`}},"../utils/lib/capitalizeFirstLetter.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.capitalizeFirstLetter=void 0;e.capitalizeFirstLetter=t=>t.charAt(0).toUpperCase()+t.slice(1)},"../utils/lib/countBytesInString.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.countBytesInString=void 0;e.countBytesInString=t=>encodeURI(t).split(/%..|./).length-1},"../utils/lib/createDeferred.js":(t,e,r)=>{"use strict";var n=r("../../node_modules/es6-promise/dist/es6-promise.js").Promise;Object.defineProperty(e,"__esModule",{value:!0}),e.createDeferred=void 0;e.createDeferred=t=>{let e=()=>{},r=()=>{};const i=new n(((t,n)=>{e=t,r=n}));return{id:t,resolve:e,reject:r,promise:i}}},"../utils/lib/createTimeoutPromise.js":(t,e,r)=>{"use strict";var n=r("../../node_modules/es6-promise/dist/es6-promise.js").Promise;Object.defineProperty(e,"__esModule",{value:!0}),e.createTimeoutPromise=void 0;e.createTimeoutPromise=t=>new n((e=>setTimeout(e,t)))},"../utils/lib/getNumberFromPixelString.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.getNumberFromPixelString=void 0;e.getNumberFromPixelString=t=>parseInt(t.replace("px",""),10)},"../utils/lib/getRandomNumberInRange.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.getRandomNumberInRange=void 0;e.getRandomNumberInRange=(t,e)=>Math.floor(Math.random()*(e-t+1))+t},"../utils/lib/getWeakRandomId.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.getWeakRandomId=void 0;e.getWeakRandomId=t=>{let e="";const r="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";for(let n=0;n<t;n++)e+=r.charAt(Math.floor(Math.random()*r.length));return e}},"../utils/lib/hasUppercaseLetter.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.hasUppercaseLetter=void 0;const r=new RegExp("^(.*[A-Z].*)$");e.hasUppercaseLetter=t=>r.test(t)},"../utils/lib/index.js":function(t,e,r){"use strict";var n=this&&this.__createBinding||(Object.create?function(t,e,r,n){void 0===n&&(n=r);var i=Object.getOwnPropertyDescriptor(e,r);i&&!("get"in i?!e.__esModule:i.writable||i.configurable)||(i={enumerable:!0,get:function(){return e[r]}}),Object.defineProperty(t,n,i)}:function(t,e,r,n){void 0===n&&(n=r),t[n]=e[r]}),i=this&&this.__setModuleDefault||(Object.create?function(t,e){Object.defineProperty(t,"default",{enumerable:!0,value:e})}:function(t,e){t.default=e}),o=this&&this.__exportStar||function(t,e){for(var r in t)"default"===r||Object.prototype.hasOwnProperty.call(e,r)||n(e,t,r)},s=this&&this.__importStar||function(t){if(t&&t.__esModule)return t;var e={};if(null!=t)for(var r in t)"default"!==r&&Object.prototype.hasOwnProperty.call(t,r)&&n(e,t,r);return i(e,t),e};Object.defineProperty(e,"__esModule",{value:!0}),e.xssFilters=e.versionUtils=e.bufferUtils=void 0,o(r("../utils/lib/createDeferred.js"),e),o(r("../utils/lib/arrayDistinct.js"),e),o(r("../utils/lib/arrayPartition.js"),e),o(r("../utils/lib/arrayToDictionary.js"),e),e.bufferUtils=s(r("../utils/lib/bufferUtils.js")),o(r("../utils/lib/bytesToHumanReadable.js"),e),o(r("../utils/lib/mergeObject.js"),e),o(r("../utils/lib/getWeakRandomId.js"),e),o(r("../utils/lib/getRandomNumberInRange.js"),e),o(r("../utils/lib/capitalizeFirstLetter.js"),e),o(r("../utils/lib/countBytesInString.js"),e),o(r("../utils/lib/truncateMiddle.js"),e),o(r("../utils/lib/getNumberFromPixelString.js"),e),o(r("../utils/lib/isUrl.js"),e),o(r("../utils/lib/hasUppercaseLetter.js"),e),o(r("../utils/lib/isAscii.js"),e),o(r("../utils/lib/isNotUndefined.js"),e),e.versionUtils=s(r("../utils/lib/versionUtils.js")),o(r("../utils/lib/objectPartition.js"),e),o(r("../utils/lib/parseHostname.js"),e),o(r("../utils/lib/throwError.js"),e),o(r("../utils/lib/isHex.js"),e),o(r("../utils/lib/resolveStaticPath.js"),e),o(r("../utils/lib/createTimeoutPromise.js"),e),o(r("../utils/lib/redactUserPath.js"),e),e.xssFilters=s(r("../utils/lib/xssFilters.js"))},"../utils/lib/isAscii.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.isAscii=void 0,e.isAscii=function(t){return!t||/^[\x00-\x7F]*$/.test(t)}},"../utils/lib/isHex.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.isHex=void 0;e.isHex=t=>/^(0x|0X)?[0-9A-Fa-f]+$/g.test(t)},"../utils/lib/isNotUndefined.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.isNotUndefined=void 0;e.isNotUndefined=t=>void 0!==t},"../utils/lib/isUrl.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.isUrl=void 0;const r=/^(http|ws)s?:\/\/[a-z0-9]([a-z0-9.-]+)?(:[0-9]{1,5})?((\/)?(([a-z0-9-_])+(\/)?)+)$/i;e.isUrl=t=>r.test(t)},"../utils/lib/mergeObject.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.mergeObject=void 0;const r=t=>t&&"object"==typeof t&&!Array.isArray(t);e.mergeObject=(t,n)=>Object.entries(n).reduce(((t,[n,i])=>{const o=t[n];return Object.assign(Object.assign({},t),{[n]:r(o)&&r(i)?(0,e.mergeObject)(o,i):i})}),t)},"../utils/lib/objectPartition.js":function(t,e){"use strict";var r=this&&this.__rest||function(t,e){var r={};for(var n in t)Object.prototype.hasOwnProperty.call(t,n)&&e.indexOf(n)<0&&(r[n]=t[n]);if(null!=t&&"function"==typeof Object.getOwnPropertySymbols){var i=0;for(n=Object.getOwnPropertySymbols(t);i<n.length;i++)e.indexOf(n[i])<0&&Object.prototype.propertyIsEnumerable.call(t,n[i])&&(r[n[i]]=t[n[i]])}return r};Object.defineProperty(e,"__esModule",{value:!0}),e.objectPartition=void 0;e.objectPartition=(t,e)=>e.reduce((([t,e],n)=>{const i=e,o=n,s=i[o],u=r(i,["symbol"==typeof o?o:o+""]);return void 0!==s?[Object.assign(Object.assign({},t),{[n]:s}),u]:[t,e]}),[{},t])},"../utils/lib/parseHostname.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.parseHostname=void 0;const r=/^([a-z0-9.+-]+:\/\/)?([a-z0-9.-]+)([:/][^:/]+)*\/?$/i;e.parseHostname=t=>{var e,n;return null===(n=null===(e=t.match(r))||void 0===e?void 0:e[2])||void 0===n?void 0:n.toLowerCase()}},"../utils/lib/redactUserPath.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.redactUserPathFromString=e.startOfUserPathRegex=void 0,e.startOfUserPathRegex=/([/\\][Uu]sers[/\\]{1,4})([^"^'^[^\]^/^\\]*)/g;e.redactUserPathFromString=t=>t.replace(e.startOfUserPathRegex,"$1[*]")},"../utils/lib/resolveStaticPath.js":(t,e,r)=>{"use strict";var n=r("../../node_modules/process/browser.js");Object.defineProperty(e,"__esModule",{value:!0}),e.resolveStaticPath=void 0;e.resolveStaticPath=(t,e=n.env.ASSET_PREFIX)=>`${e||""}/static/${t.replace(/^\/+/,"")}`},"../utils/lib/throwError.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.throwError=void 0;e.throwError=t=>{throw new Error(t)}},"../utils/lib/truncateMiddle.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.truncateMiddle=void 0;e.truncateMiddle=(t,e,r)=>{if(t.length<=e+r)return t;return`${t.substring(0,e)}…${t.substring(t.length-r,t.length)}`}},"../utils/lib/versionUtils.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.normalizeVersion=e.isNewerOrEqual=e.isEqual=e.isNewer=e.isVersionArray=void 0;const r=t=>({major:t[0],minor:t[1],patch:t[2]}),n=t=>{const e=t.split(".");if(3!==e.length)throw new Error("version string is in wrong format");return e},i=t=>`${t[0]}.${t[1]}.${t[2]}`;e.isVersionArray=t=>3===t.length;e.isNewer=(t,e)=>{const i=r("string"==typeof t?n(t):t),o=r("string"==typeof e?n(e):e);return i.major-o.major!=0?i.major>o.major:i.minor-o.minor!=0?i.minor>o.minor:i.patch-o.patch!=0&&i.patch>o.patch};e.isEqual=(t,e)=>("string"==typeof t?t:i(t))===("string"==typeof e?e:i(e));e.isNewerOrEqual=(t,r)=>(0,e.isNewer)(t,r)||(0,e.isEqual)(t,r);e.normalizeVersion=t=>t.replace(/\b0+(\d)/g,"$1")},"../utils/lib/xssFilters.js":(t,e)=>{"use strict";Object.defineProperty(e,"__esModule",{value:!0}),e.inDoubleQuotes=e.inSingleQuotes=e.inHTML=void 0;const r=/</g,n=/'/g,i=/"/g;e.inHTML=t=>t.replace(r,"&lt;");e.inSingleQuotes=t=>t.replace(n,"&#39;");e.inDoubleQuotes=t=>t.replace(i,"&quot;")},"../../node_modules/base64-js/index.js":(t,e)=>{"use strict";e.byteLength=function(t){var e=a(t),r=e[0],n=e[1];return 3*(r+n)/4-n},e.toByteArray=function(t){var e,r,o=a(t),s=o[0],u=o[1],f=new i(function(t,e,r){return 3*(e+r)/4-r}(0,s,u)),l=0,c=u>0?s-4:s;for(r=0;r<c;r+=4)e=n[t.charCodeAt(r)]<<18|n[t.charCodeAt(r+1)]<<12|n[t.charCodeAt(r+2)]<<6|n[t.charCodeAt(r+3)],f[l++]=e>>16&255,f[l++]=e>>8&255,f[l++]=255&e;2===u&&(e=n[t.charCodeAt(r)]<<2|n[t.charCodeAt(r+1)]>>4,f[l++]=255&e);1===u&&(e=n[t.charCodeAt(r)]<<10|n[t.charCodeAt(r+1)]<<4|n[t.charCodeAt(r+2)]>>2,f[l++]=e>>8&255,f[l++]=255&e);return f},e.fromByteArray=function(t){for(var e,n=t.length,i=n%3,o=[],s=16383,u=0,a=n-i;u<a;u+=s)o.push(f(t,u,u+s>a?a:u+s));1===i?(e=t[n-1],o.push(r[e>>2]+r[e<<4&63]+"==")):2===i&&(e=(t[n-2]<<8)+t[n-1],o.push(r[e>>10]+r[e>>4&63]+r[e<<2&63]+"="));return o.join("")};for(var r=[],n=[],i="undefined"!=typeof Uint8Array?Uint8Array:Array,o="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/",s=0,u=o.length;s<u;++s)r[s]=o[s],n[o.charCodeAt(s)]=s;function a(t){var e=t.length;if(e%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var r=t.indexOf("=");return-1===r&&(r=e),[r,r===e?0:4-r%4]}function f(t,e,n){for(var i,o,s=[],u=e;u<n;u+=3)i=(t[u]<<16&16711680)+(t[u+1]<<8&65280)+(255&t[u+2]),s.push(r[(o=i)>>18&63]+r[o>>12&63]+r[o>>6&63]+r[63&o]);return s.join("")}n["-".charCodeAt(0)]=62,n["_".charCodeAt(0)]=63},"../../node_modules/buffer/index.js":(t,e,r)=>{"use strict";var n=r("../../node_modules/base64-js/index.js"),i=r("../../node_modules/ieee754/index.js"),o="function"==typeof Symbol&&"function"==typeof Symbol.for?Symbol.for("nodejs.util.inspect.custom"):null;
```

using import from /lib/specificFunction bundles only this specific function

```
import * as bufferUtils from '@trezor/utils/lib/bufferUtils';

console.log(bufferUtils);
```

result:
```
(()=>{var t={"../utils/lib/bufferUtils.js":(t,r,e)=>{"use strict";var Buffer=e("../../node_modules/buffer/index.js").lW;Object.defineProperty(r,"__esModule",{value:!0}),r.reverseBuffer=void 0;r.reverseBuffer=t=>{if(t.length<1)return t;const r=Buffer.alloc(t.length);let e=r.length-1;for(let n=0;n<r.length/2;n++)r[n]=t[e],r[e]=t[n],e--;return r}},"../../node_modules/base64-js/index.js":(t,r)=>{"use strict";r.byteLength=function(t){var r=s(t),e=r[0],n=r[1];return 3*(e+n)/4-n},r.toByteArray=function(t){var r,e,i=s(t),f=i[0],u=i[1],a=new o(function(t,r,e){return 3*(r+e)/4-e}(0,f,u)),h=0,p=u>0?f-4:f;for(e=0;e<p;e+=4)r=n[t.charCodeAt(e)]<<18|n[t.charCodeAt(e+1)]<<12|n[t.charCodeAt(e+2)]<<6|n[t.charCodeAt(e+3)],a[h++]=r>>16&255,a[h++]=r>>8&255,a[h++]=255&r;2===u&&(r=n[t.charCodeAt(e)]<<2|n[t.charCodeAt(e+1)]>>4,a[h++]=255&r);1===u&&(r=n[t.charCodeAt(e)]<<10|n[t.charCodeAt(e+1)]<<4|n[t.charCodeAt(e+2)]>>2,a[h++]=r>>8&255,a[h++]=255&r);return a},r.fromByteArray=function(t){for(var r,n=t.length,o=n%3,i=[],f=16383,u=0,s=n-o;u<s;u+=f)i.push(a(t,u,u+f>s?s:u+f));1===o?(r=t[n-1],i.push(e[r>>2]+e[r<<4&63]+"==")):2===o&&(r=(t[n-2]<<8)+t[n-1],i.push(e[r>>10]+e[r>>4&63]+e[r<<2&63]+"="));return i.join("")};for(var e=[],n=[],o="undefined"!=typeof Uint8Array?Uint8Array:Array,i="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/",f=0,u=i.length;f<u;++f)e[f]=i[f],n[i.charCodeAt(f)]=f;function s(t){var r=t.length;if(r%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var e=t.indexOf("=");return-1===e&&(e=r),[e,e===r?0:4-e%4]}function a(t,r,n){for(var o,i,f=[],u=r;u<n;u+=3)o=(t[u]<<16&16711680)+(t[u+1]<<8&65280)+(255&t[u+2]),f.push(e[(i=o)>>18&63]+e[i>>12&63]+e[i>>6&63]+e[63&i]);return f.join("")}n["-".charCodeAt(0)]=62,n["_".charCodeAt(0)]=63},"../../node_modules/buffer/index.js":(t,r,e)=>{"use strict";var n=e("../../node_modules/base64-js/index.js"),o=e("../../node_modules/ieee754/index.js"),i="function"==typeof Symbol&&"function"==typeof Symbol.for?Symbol.for("nodejs.util.inspect.custom"):null;
```

Reply:
Oh I see there are two problems, tree shaking is not enabled for iframe in webpack config and libs/ are builder to commonjs which is not tree shakable anyway. So never mind I will fix this later.